src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/SetOfValueComparer.cs

```diff
@@ -17,6 +17,14 @@ internal static int Compare(ReadOnlySpan<byte> x, ReadOnlySpan<byte> y)
             int min = Math.Min(x.Length, y.Length);
             int diff;
 
+#if NET7_0_OR_GREATER
+            int diffIndex = x.CommonPrefixLength(y);
```

Comment:
Nit: `diffIndex` isn't 100% accurate; it's really `sameLength`.  For example, if it returns 0, it could be because both are empty.  Not a big deal, though.

Reply:
I'm 99% sure, based on the comment after the check/loop, that empty is never valid in context

Reply:
Even if empty the behavior is still correct, but Stephen is right - the variable name could have been a bit better. On the other hand, I didn't think it was worth burning X number of kilowatts of CI power again.